config/compression.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import "fmt"
+
+type Compression struct {
+	Request  ReqCompressionConfig  `mapstructure:"request"`
+	Response RespCompressionConfig `mapstructure:"response"`
+}
+
+// CompressionConfig defines if compression is enabled and what type of compression to use
+type ReqCompressionConfig struct {
+	Enabled bool     `mapstructure:"enabled"`
+	Kind    []string `mapstructure:"kind,flow"`
+	kindMap map[CompressionKind]bool
+}
+
+type RespCompressionConfig struct {
+	Enabled bool   `mapstructure:"enabled"`
+	Kind    string `mapstructure:"kind"`
+}
+
+type CompressionKind string
+
+const (
+	CompressionGZIP CompressionKind = "gzip"
+)
+
+func (k CompressionKind) IsValid() bool {
+	switch k {
+	// Case for valid types. As new compression types are added they should
+	// be added here as a comma separated list.
+	case CompressionGZIP:
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
Did we decide in the last code review pass to make the gzip literal checks case insensitive?

Reply:
We did and I added it for the value in `Content-Encoding` header [here](https://github.com/prebid/prebid-server/pull/2734/files#diff-785f5ab5ebde657757f3b3c34bf30a939782678d61bf1ea43f844c4f22a012d9R343) but missed it for the values in config. Will push another update with that. Good catch. Thanks!